### PR TITLE
Fix UI tests affected by UIAutomator update

### PR DESCRIPTION
**Pull request by AndiAJ**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1881349

### CODE REVIEW

**Comment by AndiAJ** [comment]
Fenix:
Failing UI tests:
- `BookmarksTest#verifySearchBookmarksViewTest`
- `ComposeBookmarksTest#verifySearchBookmarksViewTest`
- `HistoryTest#verifySearchHistoryViewTest`
- `ComposeHistoryTest#verifySearchHistoryViewTest`

All 4 failed when trying to dismiss the search bar by clicking the search_wrapper.
For unknown reasons(it works properly on main), the click on the search wrapper wasn't properly performed, so I've change the click action to a scroll action

All 4 UI tests successfully passed 25x on Firebase ✅ 

Focus:
Failing UI tests:
- `ContextMenusTest#copyLinkAddressTest `
- `ContextMenusTest#copyImageLocationTest `

The 2 UI tests failed when trying to perform the long click on the toolbar.
Switched from using `UISelector` long click to `BySelector` click with a duration.

Afterwards, constantly bumped into `RootViewWithoutFocusException` when trying to verify the context menu items.
To overcome this problem, I've added an assertion, in this way the context menu will be in focus before trying to verify the items.

Both UI tests successfully passed 25x on Firebase ✅

**Review by jjSDET — APPROVED** [review]
lgtm

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt [hunk 1]
```diff
@@ -296,7 +297,7 @@ class SearchRobot {
 
     fun tapOutsideToDismissSearchBar() {
         Log.i(TAG, "tapOutsideToDismissSearchBar: Trying to click the search wrapper")
```
[the remark this answers is not in the document]
Comment (AndiAJ) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt]: Done !

**Review by jjSDET — APPROVED** [review]
lgtm

**Review by sv-ohorvath — APPROVED** [review]

**Comment by rvandermeulen** [comment]
Folded into #5711, thanks!

**Pull request closed without merging**